### ISSUE TITLE
Warning: `NaN` is an invalid value for the `width` css style property.

### ISSUE THREAD

**Issue by nirhaas** [rpearce/react-medium-image-zoom#375]
## Issue Type

- [x] Bug report
- [ ] Feature request
- [ ] Question / support request
- [ ] Other

## Description

Hi. I have a Docusaurus website that is using this project with [ThemedImage](https://docusaurus.io/docs/markdown-features/assets#themed-images). When switching themes (dark - lite), we get the following error:

```
react-dom.development.js?ac89:67 Warning: `NaN` is an invalid value for the `width` css style property.
    at img
    at div
    at dialog
    at div
    at ControlledBase (webpack-internal:///./node_modules/react-medium-image-zoom/dist/index.js:350:9)
    at Controlled
    at Uncontrolled (webpack-internal:///./node_modules/react-medium-image-zoom/dist/index.js:649:84)
...
```

I have seen #357, and indeed it seems like ThemedImage is using `display: none` ([source](https://github.com/facebook/docusaurus/blob/8277b0bec7d4c4270f36525b8f3688c4a2a45e00/packages/docusaurus-theme-classic/src/theme/ThemedImage/styles.module.css))., but upgrading to the latest version does not seem to help. This is tested using v5.1.1.

Maybe worth noting that it was working fine on v4.4.3.

Appreciate your help.

---

Here is a code sample from my project:

Usage:
```jsx
import ThemedImage from '@theme/ThemedImage';
import useBaseUrl from '@docusaurus/useBaseUrl';
import Zoom from "react-medium-image-zoom";

...

<Zoom>
	<ThemedImage sources={{
	  light: useBaseUrl('/img/architecture-light.svg'),
	  dark: useBaseUrl('/img/architecture-dark.svg'),
	}} alt="architecture"/>
</Zoom>
```

**Comment by rpearce** [comment]
Hi, @nirhaas! I'm sorry you're having trouble. I should be able to address this tonight or tomorrow night given I have enough information here. Stay tuned!

**Comment by rpearce** [comment]
Okay, I've been able to reproduce this, and I think it directly has to do with SVGs.

I'll need some time to figure it out. Thanks for reporting this!

**Comment by arthureberle** [comment]
Have any solution been found yet? I am getting the same error right now in my app.

**Comment by rpearce** [comment]
@arthureberle I haven't fixed it yet. I'll think about it some more

**Comment by jeremyzilar** [comment]
I am having this issue as well. It seems to only happen on some of our pages that use `<Zoom>` https://preview.pousette-dartfoundation.org/works/head-of-a-woman/
- Is it possible that the file size of some of the images is too large?
- Should I try to wait until the image is fully loaded using `onLoadingComplete` before attempting to apply the Zoom?

**Comment by rpearce** [comment]
I'll try to fix this over the weekend; I know it's annoying, y'all. @jeremyzilar It looks like you're using a `.webp` image (good on ya), so I'll use that one and an SVG without dimensions in my testing.

**Comment by DamianDobrev** [comment]
I've found that for me it crashes when in the same page I have this component + a react-bootstrap Modal component. Once I open the latter - I see the same error in the console.

Is there a temporary workaround for this?

**Comment by arthureberledev** [comment]
Could fix it now by myself. The error occurred in my project because i had image components which are rendered differently on desktop and mobile. On Desktop i have used this zoom package and on mobile it was displayed as a slider. Even though the zoom component was not rendered into the DOM, it somehow was looking for a width value. 

My previous way to hide the components was with media queries through tailwindcss:
```
          <div className="block md:hidden">
            <SliderGallery images={product.images} /> // Mobile
          </div>

          <div className="hidden md:block">
            <ZoomGallery images={product.images} /> // Desktop
          </div>
```

This led to the error above. I have now switched to a use-hook called useWindowSize (which can be found on the usehooks-ts page) and a simple if then and the error dissappeared.

**Comment by rpearce** [comment]
I know I still need to fix this. We were dealing with floods and a cyclone in New Zealand, and I have other open source projects, too.

**Comment by cvanputt** [comment]
This warning is happening in 5.1.3 release for png files too:
```
<Zoom zoomMargin={8}>
  <Image
    src="https://picsum.photos/600/800"
    alt="Sample billing statement"
    width="200"
  />
</Zoom>
```

**Comment by rpearce** [comment]
Alright, nobody has submitted a PR, so this one's time has come: I'm not doing other OSS work until y'all are taken care of with this issue. TBD on a fix timeline, but here we go...

**Comment by rpearce** [comment]
Ok, I think I got it. Can y'all test this potential fix by installing `react-medium-image-zoom@5.1.4-beta.0` and seeing if that resolves your issue? If so, I can bundle this and another issue and cut a release for `v5.1.4` tomorrow.

Here's the PR: https://github.com/rpearce/react-medium-image-zoom/pull/404

**Comment by rpearce** [comment]
Can y'all please 🙏 try this fix out? `npm i react-medium-image-zoom@5.1.4-rc.0`
